src/bigwig/BigWigReader.ts

```diff
@@ -353,26 +359,88 @@ async function loadLeafNodesForRPNode(bufferedLoader: BufferedDataLoader, little
+ * 
+ * @param data Raw bed data
+ * @param filterStartChromIndex starting chromosome index used for filtering
+ * @param filterStartBase starting base used for filtering
+ * @param filterEndChromIndex ending chromosome index used for filtering
+ * @param filterEndBase ending base used for filtering
+ * @param chromDict dictionary of indices used by the file to chromosome names, conveniently stored as an array.
+ */
+export function parseBigBed(chrom: string, startBase: number, endBase: number, rest: string): BigBedData {
+    const entry: BigBedData = {
+        chr: chrom,
+        start: startBase,
+        end: endBase
+    }
+
+    let tokens = rest.split("\t");
+    if (tokens.length > 0) {
+        entry.name = tokens[0];
+    }
+    if (tokens.length > 1) {
+        entry.score = parseFloat(tokens[1]);
+    }
+    if (tokens.length > 2) {
+        entry.strand = tokens[2];
+    }
+    if (tokens.length > 3) {
+        entry.cdStart = parseInt(tokens[3]);
+    }
+    if (tokens.length > 4) {
+        entry.cdEnd = parseInt(tokens[4]);
+    }
+    if (tokens.length > 5 && tokens[5] !== "." && tokens[5] !== "0") {
+        let color: string;
+        if (tokens[5].includes(",")) {
+            color = tokens[5].startsWith("rgb") ? tokens[5] : "rgb(" + tokens[5] + ")";
+        } else {
+            color = tokens[5];
+        }
+        entry.color = color;
+    }
+    if (tokens.length > 8) {
+        const exonCount = parseInt(tokens[6]);
+        const exonSizes = tokens[7].split(',');
+        const exonStarts = tokens[8].split(',');
+        const exons: Array<BigBedExon> = [];
+
+        for (var i = 0; i < exonCount; i++) {
+            const eStart = startBase + parseInt(exonStarts[i]);
+            const eEnd = eStart + parseInt(exonSizes[i]);
+            exons.push({ start: eStart, end: eEnd });
+        }
+
+        entry.exons = exons;
+    }
+
+    return entry;
+}
 
 /**
  * Extract useful data from sections of raw big binary bed data
  * 
+ * @template T
  * @param data Raw bed data
  * @param filterStartChromIndex starting chromosome index used for filtering
  * @param filterStartBase starting base used for filtering
  * @param filterEndChromIndex ending chromosome index used for filtering
  * @param filterEndBase ending base used for filtering
  * @param chromDict dictionary of indices used by the file to chromosome names, conveniently stored as an array.
+ * @param [restParser] Parse for getting data
  */
-function decodeBedData(data: ArrayBuffer, filterStartChromIndex: number, filterStartBase: number, filterEndChromIndex: number,
-        filterEndBase: number, chromDict: Array<string>): Array<BigBedData> {
-    const decodedData: Array<BigBedData> = [];
+function decodeBedData<T>(data: ArrayBuffer, filterStartChromIndex: number, filterStartBase: number, filterEndChromIndex: number,
+    filterEndBase: number, chromDict: Array<string>, restParser?: ParseFunction<T>): Array<T> {
```

Comment:
I would expect that `restParser` would not be optional.

Reply:
Done